core/src/test/scala/akka/persistence/cassandra/CassandraSpec.scala

```diff
@@ -97,12 +98,37 @@ abstract class CassandraSpec(
     case Embedded => randomPort
   }
 
+  def keyspaces(): Set[String] = Set(journalName, snapshotName)
+
+  private var failed = false
+
+  override protected def withFixture(test: NoArgTest): Outcome = {
+    val out = super.withFixture(test)
+    if (!out.isSucceeded)
+      failed = true
+
+    out
+  }
   override protected def externalCassandraCleanup(): Unit =
     Try {
-      system.log.info(s"Dropping keysapces: $journalName $snapshotName")
-      cluster.connect().execute(s"drop keyspace if exists $journalName")
-      cluster.connect().execute(s"drop keyspace if exists $snapshotName")
-      cluster.close()
+      val c = cluster.connect(journalName)
```

Comment:
this dumps the rows if the tests fails, has been very useful for debugging build failures.